### DISCUSSION TITLE
How to add select2_from_array to Table-pro Field for Row?

### DISCUSSION THREAD

**Discussion by ngthanhluan1505** [Laravel-Backpack/community-forum#779] [Q&A (Help)]
I use Table-pro to create table for Create Operation. I need add select2_from_array to table but it can only add text field. How to fix it?
```php
CRUD::addField([
            'name' => 'table_event',
            'label' => 'Events',
            'type' => 'table',
            'entity_singular' => '　', // used on the "Add X" button
            'columns' => [
                'event_name' => 'Name', // Text
                'detail' => 'Detail',   // Text
                'type' => 'Event type', // I would like to add select2_from_array
            ],
            'max' => 10, // maximum rows allowed in the table
            'min' => 1, // minimum rows allowed in the table
        ]);
```
Note: I can use Repeatable but that design is not suitable for my function => So, Can use Table-pro with some fields?

**Comment by phpfour** [comment]
Hi @ngthanhluan1505,

Thanks for your question. Unfortunately, at this moment we are only supporting text fields in the Table field, as can be seen in the source code (`vendor/backpack/pro/resources/views/fields/table.blade.php`): 

![CleanShot 2023-12-05 at 21 13 06](https://github.com/Laravel-Backpack/community-forum/assets/171715/8b7e0162-9bde-46f2-89f2-fd3a3dbe52e1)

You can create a custom field based on the above file and add your required functionality. Here's how you can get started: 

https://backpackforlaravel.com/docs/6.x/crud-fields#creating-a-custom-field-type-1

Hope this helps.
